### DISCUSSION TITLE
SERVERPOD POSTGRES PROBLEM

### DISCUSSION THREAD

**Discussion by DD00M** [serverpod/serverpod#1523] [General]
Hi! I have a serverpod server and i want to create some tables into the postgres database from port 8090. After i create the file in protocol/file.yaml and execute serverpod generate and create the endpoint that i will access, i access it through postman.
this is the yaml:
class: Test
table: test
fields:
  title: String
  content: String
  
  and this is the endpoint class:
  
  import 'package:serverpod/serverpod.dart';
import 'package:test_bd_server/src/generated/protocol.dart';

class TestEndpoint extends Endpoint {
  Future<List<Test>> getTests(Session session, {String? keyword}) async {
    final elem = await Test.find(session,
        where: (t) =>
            keyword != null ? t.title.like('%$keyword') : Constant(true));

    return elem;
  }

  Future<bool> addTest(
      Session session, String arg_title, String arg_content) async {
    Test test = Test(content: arg_content, title: arg_title);
    Test.insert(session, test);
    return true;
  }
}

this is the postman GET request:

{
    "method": "addTest",
    "arg_title": "titlu_test",
    "arg_content": "content_test"
}

after this request executes, i get this error in server:

2023-11-03 07:44:47.840836Z Internal server error. Zoned exception.
PostgreSQLSeverity.error 42P01: relation "test" does not exist
package:postgres/src/connection.dart 520:18                     _PostgreSQLExecutionContextMixin._query
package:postgres/src/connection.dart 481:7                      _PostgreSQLExecutionContextMixin.query
package:postgres_pool/postgres_pool.dart 801:23                 _PgExecutionContextWrapper.query.<fn>
package:postgres_pool/postgres_pool.dart 800:12                 _PgExecutionContextWrapper.query
package:postgres_pool/postgres_pool.dart 615:16                 PgPool.query.<fn>
package:postgres_pool/postgres_pool.dart 352:22                 PgPool.run.<fn>.<fn>
package:postgres_pool/postgres_pool.dart 451:27                 PgPool._useOrCreate
package:postgres_pool/postgres_pool.dart 423:14                 PgPool._withConnection.<fn>
package:executor/src/executor_impl.dart 61:19                   _Executor.scheduleTask
package:postgres_pool/postgres_pool.dart 351:18                 PgPool.run.<fn>
package:retry/retry.dart 131:16                                 RetryOptions.retry
package:postgres_pool/postgres_pool.dart 349:14                 PgPool.run
package:serverpod/src/database/database_connection.dart 363:11  DatabaseConnection.insert
package:serverpod/src/database/database.dart 137:5              Database.insert
===== asynchronous gap ===========================
package:postgres_pool/postgres_pool.dart 351:18                 PgPool.run.<fn>
package:retry/retry.dart 131:16                                 RetryOptions.retry
package:postgres_pool/postgres_pool.dart 349:14                 PgPool.run
package:serverpod/src/database/database_connection.dart 363:11  DatabaseConnection.insert
package:serverpod/src/database/database.dart 137:5              Database.insert

what am i doing wrong? where is the mistake? in postman i get "true" after i execute this. Help me please :).

**Comment by SandPod** [comment]
Hey! 👋 

This is a common issue when the database has not been migrated to the latest schema. The message `PostgreSQLSeverity.error 42P01: relation "test" does not exist` informs us that the `test` table doesn't exist in the database.

Try this out this solution: https://stackoverflow.com/questions/76180598/serverpod-sql-error-when-starting-a-clean-project

With the next version of Serverpod we will add warnings so that you are informed when there is a missmatch between the connected database and the running Serverpod server.
